assets/questions/mostCommonChar.js

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Question data for Most Common Character.
+ */
+
+globalData.questions['common'] = {  // eslint-disable-line dot-notation
+  title: 'Most Common Character',
+  starterCode: {
+    python:
+`def findMostCommonChar(s):
+    return ""
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+    @classmethod
+    def foo(cls, word):
+        return None
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'Find the most common character of a string, as described in title. ',
+        "Assume that the string is ASCII, and there's only one most common character. ",
+        'Consider that the answer is unique for now.'
+      ].join(''),
+      [
+        'We will guarantee that you will always be passed a string, so ',
+        "you don't have to take care of integrity check."
+      ].join('')
+    ],
+    prerequisiteSkills: ['String', 'String Manipulation', 'Array'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'findMostCommonChar',
+    correctnessTests: [{
+      input: '',
+      allowedOutputs: ['']
+    }, {
+      input: 'apoiuytrewqsdf*&^%$#ba',
+      allowedOutputs: ['a']
+    }, {
+      input: 'abb4562873ba',
+      allowedOutputs: ['b']
+    }, {
+      input: 'babaabb',
+      allowedOutputs: ['b']
+    }],
+    buggyOutputTests: [],
+    performanceTests: []
+  }, {
+    instructions: [
+      'What if only 6 characters or less can possibly occur in the string?'
```

Comment:
If array is used in the task 1, then the space cost is 256 (Size of ASCII code). We can use hash map to reduce the space cost from 256 to 6 if there are at most 6 characters. This lead user to use hash map instead of array.